pkg/steps/bundle_source.go

```diff
@@ -0,0 +1,152 @@
+	"strings"
+
+	buildapi "github.com/openshift/api/build/v1"
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/results"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	coreapi "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type bundleSourceStep struct {
+	config      api.BundleSourceStepConfiguration
+	resources   api.ResourceConfiguration
+	buildClient BuildClient
+	imageClient imageclientset.ImageStreamsGetter
+	istClient   imageclientset.ImageStreamTagsGetter
+	jobSpec     *api.JobSpec
+	artifactDir string
+	dryLogger   *DryLogger
+	pullSecret  *coreapi.Secret
+}
+
+func (s *bundleSourceStep) Inputs(dry bool) (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *bundleSourceStep) Run(ctx context.Context, dry bool) error {
+	return results.ForReason("building_bundle_source").ForError(s.run(ctx, dry))
+}
+
+func (s *bundleSourceStep) run(ctx context.Context, dry bool) error {
+	source := fmt.Sprintf("%s:%s", api.PipelineImageStream, api.PipelineImageStreamTagReferenceSource)
+	var workingDir string
+	if dry {
+		workingDir = "dry-fake"
+	} else {
+		var err error
+		workingDir, err = getWorkingDir(s.istClient, source, s.jobSpec.Namespace())
+		if err != nil {
+			return fmt.Errorf("failed to get workingDir: %v", err)
+		}
+	}
+	dockerfile, err := s.bundleSourceDockerfile(dry)
+	if err != nil {
+		return err
+	}
+	build := buildFromSource(
+		s.jobSpec, api.PipelineImageStreamTagReferenceSource, s.config.To,
+		buildapi.BuildSource{
+			Type:       buildapi.BuildSourceDockerfile,
+			Dockerfile: &dockerfile,
+			Images: []buildapi.ImageSource{
+				{
+					From: coreapi.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: source,
+					},
+					Paths: []buildapi.ImageSourcePath{{
+						SourcePath:     fmt.Sprintf("%s/%s/.", workingDir, s.config.ContextDir),
+						DestinationDir: ".",
+					}},
+				},
+			},
+		},
+		"",
+		s.resources,
+		s.pullSecret,
+	)
+	return handleBuild(ctx, s.buildClient, build, dry, s.artifactDir, s.dryLogger)
+}
+
+func replaceCommand(manifestDir, pullSpec, with string) string {
+	return fmt.Sprintf("find %s -type f -exec sed -i 's?%s?%s?g' {} +", manifestDir, pullSpec, with)
+}
+
+func (s *bundleSourceStep) bundleSourceDockerfile(dry bool) (string, error) {
+	var dockerCommands []string
+	dockerCommands = append(dockerCommands, "")
+	dockerCommands = append(dockerCommands, fmt.Sprintf("FROM %s:%s", api.PipelineImageStream, api.PipelineImageStreamTagReferenceSource))
+	manifestDir := filepath.Join(s.config.ContextDir, s.config.OperatorManifests)
+	for _, sub := range s.config.Substitute {
+		replaceSpec, err := s.getFullPullSpec(sub.With, dry)
+		if err != nil {
+			return "", fmt.Errorf("failed to get replacement imagestream for image tag `%s`", sub.With)
+		}
+		dockerCommands = append(dockerCommands, fmt.Sprintf(`CMD ["bash", "-c", "%s"]`, replaceCommand(manifestDir, sub.PullSpec, replaceSpec)))
```

Comment:
Err... should this be a `RUN` instead of a `CMD`? :trollface: 

I couldn't get the modified bundle, was checking all the paths, context dirs, strings... But this looks like it could be a culprit :facepalm:

Reply:
Yup, that's right.

Fixed that and ran a docker build locally to see if the files in the image look right. It should be fine now.